src/base/use-authorized-subscription.tsx

```diff
@@ -0,0 +1,26 @@
+import { useRef } from "react";
+import { TSocketSubscribableEndpointNames } from "../types/api.types";
+import { useAuthData } from "./use-context-hooks";
+import { useSubscribe } from "./use-subscribe";
+
+export const useAuthorizedSubscription = <T extends TSocketSubscribableEndpointNames>(name: T) => {
+    const { isAuthorized, activeLoginid } = useAuthData();
+    const authTimeoutId = useRef<ReturnType<typeof setTimeout>>();
+    const { subscribe, ...rest } = useSubscribe(name);
+    const AUTH_TIMEOUT_DURATION = 400;
```

Comment:
```suggestion
    const AUTH_TIMEOUT_DURATION = 30000;
```
generally this would be safe